### DISCUSSION TITLE
Accounts/Teams

### DISCUSSION THREAD

**Discussion by joshnuss** [supabase/supabase#756] [Feature Requests]
It might be worth having the concept of "Accounts", in addition to "Users".

It would make it possible to have teams of users for SaaS apps (multiple users per account) and still be able to use RLS.

```sql
CREATE POLICY "Any team member can view documents." ON documents FOR SELECT
    USING (account_id = auth.account_id());

CREATE POLICY "Only document owner can delete a document" ON documents FOR DELETE
    USING (user_id = auth.uid());
```

Though, it may complicate things a bit, for example, what happens if a user has access to multiple accounts? Maybe the policy checks an array of `account_ids`, ie `CREATE POLICY ... USING (account_id in auth.account_ids())`

Caveat: It may create noise for apps that only need one level of identity. Though this kind of feature is tricky to bolt on later.

**Comment by silentworks** [comment]
Couldn't you create this now as it is in your own app using private tables and a policy with a SELECT query? or are you saying it would be better as an out of the box solution provided by the supabase team?

**Reply by joshnuss** [reply]
You're totally right, it could be done as a custom function in the policy.

Just pointing out that support for multiple users under a team might be a common thing (or maybe just for me).

I'm gonna try to implement it using private tables for a future project, and I'll report back. 

Cheers @silentworks!
